test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeLogs.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public static class EdgeLogs
+    {
+        public static async Task<IEnumerable<string>> CollectAsync(DateTime testStartTime, string filePrefix, CancellationToken token)
+        {
+            var paths = new List<string>();
+
+            // Save module logs
+            string[] output = await Process.RunAsync("iotedge", "list", token);
```

Comment:
Shouldn't the test know the list of modules it needs the logs for?

Reply:
Yeah, the test does. But this is down in the common library. Saving the logs happens automatically within the library; the tests only need to spell out which modules they need when they build their config.